Frogies-Project/Assets/Scripts/Core/Player/EntityBrain.cs

```diff
@@ -28,38 +25,49 @@ public class PlayerBrain
 
         public StatsController StatsController => _statsController;
 
-        public PlayerBrain(MovementData movementData, AttacksData attacksData, IMovementInputProvider inputMoveProvider, IFightingInputProvider inputFightingInputProvider, DirectionalMover mover, BasicAttacker attacker, PlayerAnimationController animation, StatsStorage statsStorage)
+        public PlayerBrain(MovementData movementData, AttacksData attacksData, IMovementInputProvider inputMoveProvider, IFightingInputProvider inputFightingInputProvider, DirectionalMover mover, PlayerAnimationController animation, StatsStorage statsStorage)
         {
             _movementData = movementData;
             _attacksData = attacksData;
             _inputMoveProvider = inputMoveProvider;
             _inputFightingInputProvider = inputFightingInputProvider;
             _mover = mover;
-            _attacker = attacker;
+            
             _animation = animation;
             var stats = statsStorage.Stats.Select(stat => stat.GetCopy()).ToDictionary(stat => stat);
             _statsController = new StatsController(stats);
             _healthSystem = new HealthSystem(_statsController);
             _enduranceSystem = new EnduranceSystem(_statsController);
+            _attacker = new BasicAttacker(_enduranceSystem);
+            zero.X = 0;
         }
 
+        private MovementInput zero = new MovementInput(){X = 0};
```

Comment:
fix ordering in the class, private variables before the methods.
code style: https://docs.google.com/document/d/1mW82LuwZpcfyC6fTewTAk7VhM1lRAoa95EEyRC26ZGM/edit